File: medplum-java/medplum-model/src/main/java/com/fhir/Project.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableProject.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Project {
  Optional<Reference> defaultPatientAccessPolicy();

  Optional<String> resourceType();

  Optional<code> language();

  Optional<Meta> meta();

  Reference owner();

  Optional<uri> implicitRules();

  Optional<String> description();

  Optional<List<code>> features();

  Optional<id> id();

  String name();

  static ImmutableProject.OwnerBuildStage builder() {
    return ImmutableProject.builder();
  }
}

File: medplum-typescript/src/model/fhir/Project.ts
import { code } from "./code";
import { id } from "./id";
import { Meta } from "./Meta";
import { Reference } from "./Reference";
import { uri } from "./uri";

export interface Project {
    resourceType: string | null | undefined;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    name: string;
    description: string | null | undefined;
    owner: Reference;
    features: code[] | null | undefined;
    defaultPatientAccessPolicy: Reference | null | undefined;
}
